### PR TITLE
[Utility] E2E Feature Path List

### LINKED ISSUE #733: [Utility] E2E Feature Path Template
## Objective

Prepare a template that can be used for driving M3 E2E feature path work.

## Origin Document

Up until now, we’ve been focused on (see references below) protocol components, but most features cross the boundaries of multiple protocols. Having E2E features paths makes it easier to define the end goal, and develop the structures, interfaces and tests before the implementation of the actual feature. Part of the work will be creating well-defined self-explanatory issues.

[E2E Feature Paths](https://www.notion.so/pocketnetwork/E2E-Feature-Paths-664c6b227d4e484993cb68033937e45c?pvs=4)

## Goals

- Outline key feature paths needed in M1
- Distribute work across protocol, backend & community
- Create a process to SPIKE the creation of these GitHub issues

## Deliverable

- [ ] A template that can be followed and filled out to scope work for E2E Feature Paths
- [ ] A document outlining the process for creating new feature path tickets
- [ ] One example of an E2E feature ticket

## Non-goals / Non-deliverables

- Exhaustive list of all the E2E feature paths (outside the scope of this ticket)

## General issue deliverables

- [ ] Update the appropriate CHANGELOG(s)
---

**Creator**: @Olshansk 
**Co-Owners**: @jessicadaugherty

### PR DESCRIPTION
**Pull request by Olshansk**
## Description

This PR introduces one new document related to end-to-end (E2E) features: `E2E_FEATURE_LIST.md`.

`E2E_FEATURE_LIST` provides an overview of E2E features split into sections: Relay, QoS, Rewards & Distributions, Burns & Slashes, PoS Functionality, Feature Flags, and Governance.

## Issue

Fixes #733

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Major breaking change
- [X] Documentation
- [ ] Other 

## List of changes

- Added the template for E2E Feature path template

### CODE REVIEW

**Review by adshmh — COMMENTED** [review]
Too early for me to propose any potential changes, but I do have a question regarding the TestNet vs MainNet on item 3, i.e. Rate Limited Applications: is there enough distinction between the two E2E tests that having the test ready for MainNet does not cover having it on TestNet?

File: utility/doc/E2E_FEATURE_LIST.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+# E2E Features <!-- omit in toc -->
+
+- [Legend](#legend)
+- [A. E2E Relay](#a-e2e-relay)
+  - [1. Trustless Relay 🟡](#1-trustless-relay-)
+  - [2. Delegated Relay 🟡](#2-delegated-relay-)
+  - [3. Rate Limited Application 🟢🔴](#3-rate-limited-application-)
```
Comment (Olshansk) [inline utility/doc/E2E_FEATURE_LIST.md:7]: @adshmh Going to open an inline thread per your comment in  https://github.com/pokt-network/pocket/pull/766#pullrequestreview-1440778055

> Too early for me to propose any potential changes, but I do have a question regarding the TestNet vs MainNet on item 3, i.e. Rate Limited Applications: is there enough distinction between the two E2E tests that having the test ready for MainNet does not cover having it on TestNet?

There's no formal definition (written down anywhere) for what we need to have in TestNet vs MainNet so it's a good question. Even between the [specification](https://github.com/pokt-network/pocket-network-protocol/tree/main/utility), the paper on [Relay Mining](https://arxiv.org/abs/2305.10672), and all the other notes we have, there are still open-ended questions w.r.t the protocol which will be answered by the future economics group.

For TestNet, we need to have all the building blocks (i.e. entrypoints) in place to validate that values are changing, communication/messaging (i.e. web2 style) is reliable and we can build upon it. E2E tests should capture behaviour. For MainNet, we'll need to make sure that everything is completely secure (i.e. no cryptographic vulnerabilities), proven (i.e. the algorithms are sound, scalable and on-chain) and we can build upon the tokenomics. In other words, TestNet -> MainNet will be the bigger hurdle.

File: utility/doc/E2E_FEATURE_LIST.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+- [B. E2E QoS](#b-e2e-qos)
```
Comment (fredteumer) [inline utility/doc/E2E_FEATURE_LIST.md:8]: cut all quality of service checks on protocol from v1 scope (imo)

File: utility/doc/E2E_FEATURE_LIST.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  - [1. Servicer Relay Rewards - Custodial 🟡](#1-servicer-relay-rewards---custodial-)
+  - [2. Servicer Relay Rewards - Non-Custodial 🟡](#2-servicer-relay-rewards---non-custodial-)
+  - [3. Servicer Relay Rewards - Rev-Share 🔵](#3-servicer-relay-rewards---rev-share-)
+  - [4. Validator - Block Rewards 🟡](#4-validator---block-rewards-)
+  - [5. Fisherman - QoS Submission Rewards 🟡](#5-fisherman---qos-submission-rewards-)
+- [D. E2E Burns \& Slashes](#d-e2e-burns--slashes)
+  - [1. Validator - Double-Sign Burn 🔴](#1-validator---double-sign-burn-)
+  - [2. Application - Application Quota Stake Burn 🟢🔴](#2-application---application-quota-stake-burn-)
+  - [3. Servicer - Inaccurate Challenged Response 🔵](#3-servicer---inaccurate-challenged-response-)
+  - [4. Fisherman - Missing Test Score Burn 🔴](#4-fisherman---missing-test-score-burn-)
```
Comment (fredteumer) [inline utility/doc/E2E_FEATURE_LIST.md:22]: cut

File: utility/doc/E2E_FEATURE_LIST.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+- [E. E2E PoS Functionality](#e-e2e-pos-functionality)
+  - [1. Actor Staking 🟡](#1-actor-staking-)
+  - [2. Actor Unstaking 🟡](#2-actor-unstaking-)
+  - [3. Actor Unbonding 🟡](#3-actor-unbonding-)
+  - [4. Actor Jailing 🟡](#4-actor-jailing-)
+  - [5. Actor Unjailing 🟡](#5-actor-unjailing-)
+  - [6. Actor Pausing 🔵](#6-actor-pausing-)
+  - [7. Actor Unpausing 🔵](#7-actor-unpausing-)
+  - [8. Delegated Actor Staking 🔵](#8-delegated-actor-staking-)
+- [F. E2E Feature Flags](#f-e2e-feature-flags)
```
Comment (fredteumer) [inline utility/doc/E2E_FEATURE_LIST.md:33]: cut from scope , can always introduce post launch

File: utility/doc/E2E_FEATURE_LIST.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  - [2. Delegated Relay 🟡](#2-delegated-relay-)
+  - [3. Rate Limited Application 🟢🔴](#3-rate-limited-application-)
+- [B. E2E QoS](#b-e2e-qos)
+  - [1. QoS - Single Check 🟡](#1-qos---single-check-)
+  - [2. QoS - Test Score Submission 🟢🔴](#2-qos---test-score-submission-)
+  - [3. QoS - Report Card Aggregation 🟢🔴](#3-qos---report-card-aggregation-)
+- [C. E2E Rewards \& Distributions](#c-e2e-rewards--distributions)
+  - [1. Servicer Relay Rewards - Custodial 🟡](#1-servicer-relay-rewards---custodial-)
+  - [2. Servicer Relay Rewards - Non-Custodial 🟡](#2-servicer-relay-rewards---non-custodial-)
+  - [3. Servicer Relay Rewards - Rev-Share 🔵](#3-servicer-relay-rewards---rev-share-)
```
Comment (fredteumer) [inline utility/doc/E2E_FEATURE_LIST.md:15]: revshare is another cut from scope option, consider it

**Review by fredteumer — COMMENTED** [review]
we're at a point in the delivery cycle where i believe we should consider making serious concessions. this is of course, my opinion, but we essentially did parity+ on V2 portal and have found great success and gains can be made. 

once we lay a strong foundation, we can revisit any of these topics and implement quickly. 

my 2 pokt

File: utility/doc/E2E_FEATURE_LIST.md (hunk 2 above)
Comment (Olshansk) [reply utility/doc/E2E_FEATURE_LIST.md:8]: Which of these would you recommend:

1. Make them nice-to-haves
2. Remove them completely from this dock
3. Add a post-launch emoji and mark it as that

Comment (fredteumer) [reply utility/doc/E2E_FEATURE_LIST.md:8]: i like option 3, let's push it to v1.X. i think it's important to continue to shape the roadmap and let everyone know that's the goal, just that we need to get to V1 faster and then iterate

File: utility/doc/E2E_FEATURE_LIST.md (hunk 4 above)
Comment (Olshansk) [reply utility/doc/E2E_FEATURE_LIST.md:33]: Pushing back on this. We have this in v0 and what we need to safely upgrade the protocol and introduce new feature.

A must-have for MainNet launch

**Comment by Olshansk** [comment]
@fredteumer Bump on this PR in case it got lost in your backlog 🙏

**Comment by fredteumer** [comment]
latest changes look fantastic and very realistic. Let's go build it now 🖖

**Pull request merged**